### PR TITLE
lib/popcorn: introduce BoolValidator in popcorn::pop_validation

### PR DESCRIPTION
**Pull request by Morriar**
A Boolean validator.

Signed-off-by: Alexandre Terrasa <alexandre@moz-code.org>

### CODE REVIEW

**Comment by Morriar** [comment]
test this please

File: lib/popcorn/pop_validation.nit [hunk 1]
```diff
@@ -516,6 +516,45 @@ class FloatField
...
 
+# Check if a field is a Bool
+#
+# ~~~
+# var validator = new ObjectValidator
+# validator.add new BoolField("field", required=false)
+# assert validator.validate("""{}""")
+# assert validator.validate("""{ "field": true }""")
+# assert validator.validate("""{ "field": false }""")
+# assert not validator.validate("""{ "field": "foo" }""")
```
Comment (privat) [inline lib/popcorn/pop_validation.nit]: maybe document that there is no type juggling.

~~~nit
assert not validator.validate("""{ "field": "true" }""")
assert not validator.validate("""{ "field": 1 }""")
assert not validator.validate("""{ "field": [true] }""")
~~~

Comment (Morriar) [inline lib/popcorn/pop_validation.nit]: Reroll: Done

File: lib/popcorn/pop_validation.nit [hunk 2]
```diff
@@ -516,6 +516,50 @@ class FloatField
...
+#
+# validator = new ObjectValidator
+# validator.add new BoolField("field")
+# assert not validator.validate("""{}""")
+# assert validator.validate("""{ "field": true }""")
+# assert validator.validate("""{ "field": false }""")
+# assert not validator.validate("""{ "field": "foo" }""")
+#
+# No type conversion is applied on the input value:
```
Comment (privat) [inline lib/popcorn/pop_validation.nit]: comments in docunits should be outside the block to be markdown rendered

~~~
# [...]
# assert not validator.validate("""{ "field": "foo" }""")
￼# ~~~
#
# No type conversion is applied on the input value:
#
# ~~~
# assert not validator.validate("""{ "field": "true" }""")

~~~

**Comment by Morriar** [comment]
Reroll with @privat requests

**Pull request merged**